pywren_ibm_cloud/tests.py

```diff
@@ -43,315 +46,362 @@
+        counter = TestMethods.my_map_function_obj(obj, 0)
+        cloudobject = internal_storage.put_object(counter)
+        return cloudobject
+
+    @staticmethod
+    def my_cloudobject_get(results, internal_storage):
+        data = [internal_storage.get_object(cloudobject) for cloudobject in results]
+        return TestMethods.my_reduce_function(data)
 
 
 class TestPywren(unittest.TestCase):
 
     def checkResult(self, result):
-        result_to_compare = STORAGE.get_object(bucket_name=STORAGE_CONFIG['bucket'],
-                                               key=f'{PREFIX}/result')
-
+        result_to_compare = STORAGE.get_object(bucket_name=STORAGE_CONFIG['bucket'], key=f'{PREFIX}/result')
         if isinstance(result, list):
             total = 0
             for r in result:
                 total += r
         else:
             total = result
-
         self.assertEqual(total, int(result_to_compare))
 
     def test_call_async(self):
         print('Testing call_async()...')
         pw = pywren.function_executor(config=CONFIG)
-        pw.call_async(hello_world, "")
+        pw.call_async(TestMethods.hello_world, "")
         result = pw.get_result()
         self.assertEqual(result, "Hello World!")
 
         pw = pywren.function_executor(config=CONFIG)
-        pw.call_async(simple_map_function, [4, 6])
+        pw.call_async(TestMethods.simple_map_function, [4, 6])
         result = pw.get_result()
         self.assertEqual(result, 10)
 
         pw = pywren.function_executor(config=CONFIG)
-        pw.call_async(simple_map_function, {'x': 2, 'y': 8})
+        pw.call_async(TestMethods.simple_map_function, {'x': 2, 'y': 8})
         result = pw.get_result()
         self.assertEqual(result, 10)
 
     def test_map(self):
         print('Testing map()...')
         iterdata = [[1, 1], [2, 2], [3, 3], [4, 4]]
         pw = pywren.function_executor(config=CONFIG)
-        pw.map(simple_map_function, iterdata)
+        pw.map(TestMethods.simple_map_function, iterdata)
+        result = pw.get_result()
+        self.assertEqual(result, [2, 4, 6, 8])
+
+        pw = pywren.function_executor(config=CONFIG, workers=1)
+        pw.map(TestMethods.simple_map_function, iterdata)
         result = pw.get_result()
         self.assertEqual(result, [2, 4, 6, 8])
 
+        pw = pywren.function_executor(config=CONFIG)
+        set_iterdata = set(range(2))
+        pw.map(TestMethods.hello_world, set_iterdata)
+        result = pw.get_result()
+        self.assertEqual(result, ['Hello World!'] * 2)
+
+        pw = pywren.function_executor(config=CONFIG)
+        generator_iterdata = range(2)
+        pw.map(TestMethods.hello_world, generator_iterdata)
+        result = pw.get_result()
+        self.assertEqual(result, ['Hello World!'] * 2)
+
+        pw = pywren.function_executor(config=CONFIG)
+        listDicts_iterdata = [{'x': 2, 'y': 8}, {'x': 2, 'y': 8}]
+        pw.call_async(TestMethods.simple_map_function, listDicts_iterdata)
```

Comment:
@omerb01 It should be pywren.map(), right?

Reply:
@JosepSampe yep, thanks